src/groups/connections.jl

```diff
@@ -0,0 +1,174 @@
+
+"""
+    AbstractCartanSchoutenConnection
+
+Abstract type for Cartan-Schouten connections, that is connections whose geodesics
+going through group identity are one-parameter subgroups. See [^Pennec2020] for details.
+
+[^Pennec2020]:
+    > X. Pennec and M. Lorenzi, “5 - Beyond Riemannian geometry: The affine connection
+    > setting for transformation groups,” in Riemannian Geometric Statistics in Medical Image
+    > Analysis, X. Pennec, S. Sommer, and T. Fletcher, Eds. Academic Press, 2020, pp. 169–229.
+    > doi: 10.1016/B978-0-12-814725-2.00012-1.
+"""
+abstract type AbstractCartanSchoutenConnection <: AbstractAffineConnection end
+
+"""
```

Comment:
It's a bit complicated. Each Lie group has at least a one-parameter family of Cartan-Schouten connections, but if the Lie algebra is commutative then the are all the same. These plus, minus and zero connections belong to this family so we could technically have a struct with a number that represents all of these. In practice these three seem to be the most important so for performance I think they should have different types.

Moreover, some groups have Cartan-Schouten connections that don't belong to this one-parameter family but I have no idea what their significance is.

Reply:
There isn't much they have in common except that they all belong to the same one-parameter family, so the improving readability of the name would be the only point of this refactoring. The problem is, `CartanSchouten{<:Symbol}` might suggest that you can get any Cartan-Schouten connection using this type but actually it would only cover these three.

Reply:
Ah right, with the last argument, you convinced me. While I still feel the names are too long, the parameter approach would yield a wrong impression maybe.